enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/upstream/UpstreamDatabaseStrategySelector.java

```diff
@@ -65,9 +66,10 @@ public MemberId bestUpstreamDatabase() throws UpstreamDatabaseSelectionException
             log.debug( "Trying selection strategy [%s]", strategy.toString() );
             try
             {
-                if ( strategy.upstreamDatabase().isPresent() )
+                Optional<MemberId> foundMemberByStrategy = strategy.upstreamDatabase();
+                if ( foundMemberByStrategy.isPresent() )
                 {
-                    result = strategy.upstreamDatabase().get();
+                    result = foundMemberByStrategy.get();
```

Comment:
Would probably be more natural as a `stream().filter(Optional::isPresent).map(Optional::get).findFirst()` call?